modules/inbucket/inbucket_test.go

```diff
@@ -0,0 +1,71 @@
+package inbucket
+
+import (
+	"context"
+	"net/smtp"
+	"testing"
+
+	"github.com/inbucket/inbucket/pkg/rest/client"
+	"github.com/testcontainers/testcontainers-go"
+)
+
+func TestInbucket(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx, testcontainers.WithImage("inbucket/inbucket:sha-2d409bb"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	// smtpConnection {
+	smtpUrl, err := container.SmtpConnection(ctx)
+	// }
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// webInterface {
+	webInterfaceUrl, err := container.WebInterface(ctx)
+	// }
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	restClient, err := client.New(webInterfaceUrl)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	headers, err := restClient.ListMailbox("to@example.org")
```

Comment:
We could make the container provide a way to retrieve the restClient, so the webInterface is not needed 🤔

Not an strong opinion on that, you are most familiar with the technology, so please consider keeping this as is.

Reply:
Mmmm forget about that. I've just realised that if the container retrieves a `github.com/inbucket/inbucket/pkg/rest/client` type, then we'll be adding that dependency to the consumers (at the moment it's only present at test time).